tests/backends/data/test_fs.py

```diff
@@ -0,0 +1,301 @@
+"""Tests for Ralph fs storage backend"""
+
+import json
+import os
+from collections.abc import Iterable
+from pathlib import Path
+from uuid import uuid4
+
+import pytest
+
+from ralph.backends.data.base import BaseOperationType, BaseQuery, DataBackendStatus
+from ralph.backends.data.fs import FileSystemDataBackend
+from ralph.conf import settings
+from ralph.exceptions import BackendParameterException
+
+
+def test_backends_data_fs_file_system_data_backend_instantiation(fs):
+    """Test the `FileSystemDataBackend` backend instantiation."""
+    # pylint: disable=invalid-name,unused-argument
+
+    assert FileSystemDataBackend.name == "fs"
+    assert FileSystemDataBackend.query_model == BaseQuery
+    assert FileSystemDataBackend.default_operation_type == BaseOperationType.CREATE
+
+    backend = FileSystemDataBackend()
+
+    assert backend.default_directory == Path(backend.settings.DEFAULT_DIRECTORY_PATH)
+    assert backend.default_query_string == backend.settings.DEFAULT_QUERY_STRING
+    assert backend.default_chunk_size == backend.settings.DEFAULT_CHUNK_SIZE
+
+    deep_path = "deep/directories/path"
+    assert not os.path.exists(deep_path)
+
+    backend = FileSystemDataBackend(
+        default_directory_path=deep_path,
+        default_query_string="foo.txt",
+        default_chunk_size=1,
+    )
+    assert os.path.exists(deep_path)
+    assert backend.default_directory == Path(deep_path)
+    assert backend.default_directory.is_dir()
+    assert backend.default_query_string == "foo.txt"
+    assert backend.default_chunk_size == 1
+
+    # Check that a storage with the same path doesn't throw an exception
+    FileSystemDataBackend(deep_path)
```

Comment:
Could you test it in a try except block to handle the exception(s) ?

Reply:
Good Idea) Thanks) Will update that)